### DISCUSSION TITLE
Compression Algorithm

### DISCUSSION THREAD

**Discussion by Giorey01** [Akylas/OSS-DocumentScanner#161] [Ideas]
I've found the size of the pdf generated too big, almost a MB each page I scan, I would suggest to integrate a compression Algorithm with 3 different flags to select the quality.
Even the exporting process takes too much time on my OPPO a94 5G, for a 27 pages pdf it takes almost a whole minute to process the pdf and export it.
The rest of the app is really nice tho, keep it up!

**Comment by farfromrefug** [comment]
@Giorey01 There is already on compression algorithm. But i agree it is not perfect.
One thing to keep in mind. If you use default settings each pdf page will be the size of the image. Your camera being 48Mp it makes quite big images. If you use a4 for pdf page format you ll that it should be much faster.

About the export being slow. Yes it is and i will try to make it faster. Right now seeing the native APIs i am using i first create PDF then create another compressed PDF. So it takes time, especially with big images.
I will try to improve all that

Thanks for the kind words

**Reply by Giorey01** [reply]
Hey thank you for your response, Right now I'm using a size of 2560x1920 and I end up with large files anyway, if I can help in some way I'll surely do that!

**Reply by farfromrefug** [reply]
@Giorey01 could you share images i could import to try and see if i can reproduce the issue?

**Reply by Giorey01** [reply]
Sure, they are just my basic Physic notes, where can I send them to you?

**Reply by Giorey01** [reply]
Here they are

https://buzzheavier.com/f/GIlCYCjOsAA=

**Reply by Giorey01** [reply]
The pdf produced by the app weights as the sum of the images, the first images are full size, some are full HD size, while the rest are 2k'ish size.

**Reply by farfromrefug** [reply]
@Giorey01 latest version should behave Much better

**Reply by Giorey01** [reply]
hey, i tried the app out and yes! it is way better.
I've managed to export a 25 pages pdf with jpeg quality set to 40 and 2k as resolution and i ended up with a 10mb file, i wanna check if playing with the jpeg quality and resolution i can get a better quality/size pdf

Keep up with your awesome work, thank you again!
